src/Aspire.Hosting.Yarp/YarpResourceExtensions.cs

```diff
@@ -38,22 +40,60 @@ public static IResourceBuilder<YarpResource> AddYarp(
+                          ctx.EnvironmentVariables["Kestrel__Certificates__Default__Password"] = ctx.Password;
+
+                          return Task.CompletedTask;
+                      });
+
+        builder.Eventing.Subscribe<BeforeStartEvent>((@event, cancellationToken) =>
+        {
+            if (!builder.ExecutionContext.IsRunMode)
+            {
+                // This configuration is only valid for run mode currently
+                return Task.CompletedTask;
+            }
+
+            var developerCertificateService = @event.Services.GetRequiredService<IDeveloperCertificateService>();
+
+            bool addHttps = false;
+            if (!resource.TryGetLastAnnotation<CertificateKeyPairAnnotation>(out var annotation))
+            {
+                if (developerCertificateService.DefaultTlsTerminationEnabled)
+                {
+                    // If no specific certificate is configured
+                    addHttps = true;
+                }
+            }
+            else if (annotation.UseDeveloperCertificate.GetValueOrDefault(developerCertificateService.DefaultTlsTerminationEnabled) || annotation.Certificate is not null)
+            {
+                addHttps = true;
+            }
+
+            if (addHttps)
+            {
+                // If a TLS certificate is configured, ensure the YARP resource has an HTTPS endpoint and
+                // configure the environment variables to use it.
+                yarpBuilder
+                    .WithHttpsEndpoint(targetPort: HttpsPort)
+                    .WithEnvironment("ASPNETCORE_HTTPS_PORT", $"{resource.GetEndpoint("https").Property(EndpointProperty.Port)}")
+                    .WithEnvironment("ASPNETCORE_URLS", $"{resource.GetEndpoint("https").Property(EndpointProperty.Scheme)}://*:{resource.GetEndpoint("https").Property(EndpointProperty.TargetPort)};{resource.GetEndpoint("http").Property(EndpointProperty.Scheme)}://*:{resource.GetEndpoint("http").Property(EndpointProperty.TargetPort)}");
+            }
+
+            return Task.CompletedTask;
+        });
 
         if (builder.ExecutionContext.IsRunMode)
         {
             yarpBuilder.WithEnvironment(ctx =>
             {
-#pragma warning disable ASPIRECERTIFICATES001 // Type is for evaluation purposes only and is subject to change or removal in future updates. Suppress this diagnostic to proceed.
-                var developerCertificateService = ctx.ExecutionContext.ServiceProvider.GetRequiredService<IDeveloperCertificateService>();
-#pragma warning restore ASPIRECERTIFICATES001 // Type is for evaluation purposes only and is subject to change or removal in future updates. Suppress this diagnostic to proceed.
-                if (!developerCertificateService.SupportsContainerTrust)
+                var options = ctx.ExecutionContext.ServiceProvider.GetRequiredService<DistributedApplicationOptions>();
+                if (options.AllowUnsecuredTransport)
```

Comment:
Seems unrelated? Its specifically about the OTLP endpoint.

Reply:
Yeah, this was a thought about the OTLP behavior, but not related to this change; I'll back it out.